spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Transformer.scala

```diff
@@ -296,3 +302,122 @@ class SampleToBatch[T: ClassTag]
     }
   }
 }
+
+/**
+ * Convert a sequence of [[Sample]] to a sequence of [[MiniBatch]] through function toMiniBatch.
+ */
+private[bigdl] class SampleToMiniBatch[T: ClassTag](
+    totalBatch: Int,
+    miniBatch: Option[MiniBatch[T]] = None,
+    paddingParam: Option[PaddingParam[T]] = None,
+    partitionNum: Option[Int] = None)
+    (implicit ev: TensorNumeric[T]) extends Transformer[Sample[T], MiniBatch[T]] {
+
+  private val batchPerPartition = Utils.getBatchSize(totalBatch)
+  var miniBatchBuffer = miniBatch.orNull
+
+  override def apply(prev: Iterator[Sample[T]]): Iterator[MiniBatch[T]] = {
+    val batchSizePerPartition = batchPerPartition
+    new Iterator[MiniBatch[T]] {
+      private val batchSize = batchSizePerPartition
+
+      private val sampleData = new Array[Sample[T]](batchSize)
```

Comment:
in the following code

```scala
sampleData(i) = sample.clone()

sampleData(i).copy(sample)
```

why we need them? as sample is immutable, why can it be copied?